### PR TITLE
Transpiler should raise when conflicting parameters, not just ignore them silently

### PR DESCRIPTION
**Pull request by 1ucian0**
If `transpile` is called with `pass_manager`, most of the additional parameters are already embed in the pass manager configuration and they are currently silently ignored.

This PR raises when that happen and includes a small refactor to reorganize the code.

### CODE REVIEW

**Comment by ajavadia** [comment]
can we just remove the `pass_manager` kwarg from `transpile(circuit, pass_manager)`? it's no different than `pass_manager.run(circuit)`. It was only included temporarily for backwards compatibility, but if you already have the passmanager you shouldn't have to call transpile. The transpile function is there to select an appropriate passmanager based on some desired transpilation criteria.

This is in the docstring:
```
        pass_manager: The pass manager to use for a custom pipeline of transpiler passes.
            If this arg is present, all other args will be ignored and the
            pass manager will be used directly (Qiskit will not attempt to
            auto-select a pass manager based on transpile options).
```

**Comment by 1ucian0** [comment]
> can we just remove the `pass_manager` kwarg from `transpile(circuit, pass_manager)`? it's no different than `pass_manager.run(circuit)`. It was only included temporarily for backwards compatibility, but if you already have the passmanager you shouldn't have to call transpile. The transpile function is there to select an appropriate passmanager based on some desired transpilation criteria.

Agree. Let's depreate the parameter in this PR too!

File: releasenotes/notes/parameter_conflict_in_transpile-fc67d76288b480c4.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+other:
+  - |
+    The function ```iqiskit.compiler.transpile``` now raises an error when conlicting parameters are passed, instead of ignoring them silently.
```
Comment (ajavadia) [inline releasenotes/notes/parameter_conflict_in_transpile-fc67d76288b480c4.yaml:4]: Can you update this note for the deprecation too? looks good otherwise

Comment (1ucian0) [reply releasenotes/notes/parameter_conflict_in_transpile-fc67d76288b480c4.yaml:4]: reno added in bab0cfaeb

**Pull request merged**